components/style/properties/properties.mako.rs

```diff
@@ -822,6 +823,17 @@ impl ToCss for PropertyDeclaration {
     }
 }
 
+impl HasViewportPercentage for PropertyDeclaration {
+    fn has_viewport_percentage(&self) -> bool {
+        match *self {
+            % for property in data.longhands:
+                PropertyDeclaration::${property.camel_case}(DeclaredValue::Value(ref val)) => val.has_viewport_percentage(),
```

Comment:
This line is too long, but wrapping this will make the generated code ugly. Suggestions welcome.